numpy/lib/_function_base_impl.py

```diff
@@ -4794,6 +4794,11 @@ def _quantile(
                                method_props, gtype)
             result_shape = virtual_indexes.shape + (1,) * (arr.ndim - 1)
             gamma = gamma.reshape(result_shape)
+            if arr.dtype.kind in ("i", "u"):
```

Comment:
nit: there is precedent for using `"iu"` instead of the tuple

Reply:
Yes, you are probably right. But first, we (or I) need to cover all branches. As pointed out above, in the current draft version, the return type still depends on the quantile `q`.

I hoped to get feedback from the maintainer if I'm on the right track, before putting in all the effort. As soon as we have the correct tests in place, we should indeed try to simplify the code. Thanks for the input.

Reply:
FWIW, I think this should be moved into `_lerp` if anything. (this brings up my long term thought that a `lerp` ufunc would maybe the cleanest thing to solve all of these issues.)
If you have it there, you can also make it quicker than just doing a full cast here I think (at least for very large arrays).  Although it may be nice to at least not do this when the input is clearly OK.

EDIT: If you are actually interested in that, at least I think that adding a `lerp` ufunc would be worthwhile.